openedx/adg/lms/webinars/admin.py

```diff
@@ -187,6 +187,22 @@ def get_queryset(self, request):
         qs = super(WebinarAdmin, self).get_queryset(request)
         return qs.filter(is_cancelled=False)
 
+    def get_deleted_objects(self, objs, request):
+        """
+        Overriding this method to prevent delete permissions error on related objects when a webinar is
+        deleted (cancelled) from admin site.
+
+        We have overridden the delete method of Webinar model to mark a webinar as cancelled, instead of deleting it.
+        Once the webinar is deleted (cancelled) from admin site, the delete view gets a list of related objects by
+        calling get_deleted_objects method and checks for admin’s delete permission of those related objects, which
+        includes webinar registrations. Since webinar registration’s delete permission is revoked from the admin site,
+        the admin delete view show a permission error and does not allow to proceed with the cancellation of the
+        webinar. To bypass the permission error, we are overriding this method.
+        """
+        deleted_objects, model_count, perms_needed, protected = \
+            super().get_deleted_objects(objs, request)  # pylint: unused-variable
+        return deleted_objects, model_count, set(), protected
```

Comment:
Do we really need to delete the registrations of cancelled webinars? We should turn off all the emails related to it but we should not delete those registrations. As that information might be useful. One scenario that comes to my mind is: 
A webinar that was created but cancelled due to some reason. In such a case, if a the webinar is created again, the admin can simply check the registrations for the cancelled webinar, and invite all the registered members.

Reply:
At the end of the docstring, can we add a small note (Note: deleting (cancelling) a webinar does not delete its corresponding webinar registrations, we only need the permission so that the webinar can be cancelled successfully).
@asadali145

Reply:
@MunamMubashir-arbisoft you mistakenly tagged Asad Ali instead of @aliassad.

Reply:
@harris-irfan thanks for the correction. Asad definitely doesn't need code related reminders this week for sure. 😛